src/runtime/CMakeLists.txt

```diff
@@ -5,6 +5,12 @@ set(HIPSYCL_RT_EXTRA_LINKER_FLAGS "")
 set(HIPSYCL_RT_SANITIZE "" CACHE STRING
     "Enable building hipSYCL runtime with specified sanitizers")
 
+list(PREPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}")
+find_package(Filesystem REQUIRED Final Experimental Boost)
+set(CXX_FILESYSTEM_HEADER "${CXX_FILESYSTEM_HEADER}" PARENT_SCOPE)
+set(CXX_FILESYSTEM_NAMESPACE "${CXX_FILESYSTEM_NAMESPACE}" PARENT_SCOPE)
+list(POP_FRONT CMAKE_MODULE_PATH)
```

Comment:
Similarly, the pop commands are not available. Can we circumvent by storing and reassigning the state of `CMAKE_MODULE_PATH` before we modify it?

Reply:
I'd rather use `REMOVE_AT`: that's one line instead of two for save/restore :)